striptease/data_interface.py

```diff
@@ -23,65 +22,65 @@ def load_sci_data(fname_or_data, module, polarimeter, detector, data_type):
-        msg = 'ERROR: type(fname) %s is wrong' % str(type(fname_or_data))
-        print(msg)
-        return -1, -1
-        
-    
-    # Module exists?
-    modules = ['O', 'Y', 'R', 'G', 'B', 'R', 'W', 'o', 'y', 'r', 'g', 'b', 'r', 'w']
-    if (module in modules) == False:
-        msg = 'ERROR: Module %s does not exist' % module
-        print(msg)
-        return -1, -1
-    module = module.upper()
-    
+    try:
+        f = open(fname_or_data)
+    except IOError as e:
+        if str(type(fname_or_data)) != "<class 'h5py._hl.dataset.Dataset'>":
+            raise ValueError(f"File can't be accessed")
+
+    # # fname of correct type?
+    # if (type(fname_or_data) != str) and (
+    #     str(type(fname_or_data)) != "<class 'h5py._hl.dataset.Dataset'>"
+    # ):
+    #     msg = "ERROR: type(fname) %s is wrong" % str(type(fname_or_data))
+    #     print(msg)
+    #     return -1, -1
+
+    # # Module exists?
+    # modules = ["O", "Y", "R", "G", "B", "R", "W", "o", "y", "r", "g", "b", "r", "w"]
+    # if not module in modules:
+    #     raise ValueError(f"Module {module} does not exist")
+
+    # module = module.upper()
+
     # Polarimeter exists?
-    polarimeters = ['0','1','2','3','4','5','6']
-    if (polarimeter in polarimeters) == False:
-        msg = 'ERROR: Polarimeter %s does not exist' % polarimeter
-        print(msg)
-        return -1, -1
-    
-    if (module+polarimeter) == 'W0':
-        msg = 'ERROR: Polarimeter %s does not exist' % module+polarimeter
-        print(msg)
-        return -1, -1
+    modules = ["O", "Y", "R", "G", "B", "R", "W"]
+    polarimeters = ["0", "1", "2", "3", "4", "5", "6"]
+    module = polarimeter[0]
+    pol    = polarimeter[1]
+
+    if (not module.upper() in modules) or (not pol.upper() in polarimeters):
+        raise ValueError(f"ERROR: Polarimeter {polarimeter} does not exist")
+
+    if polarimeter.upper() == "W0":
+        raise ValueError(f"ERROR: Polarimeter {polarimeter} does not exist")
+
+    polarimeter = polarimeter.upper()
     
     # Detector exists?
-    detectors = ['Q1', 'Q2', 'U1', 'U2']
-    if (detector in detectors) == False:
-        msg = 'ERROR: Detector %s does not exist' % detector
-        print(msg)
-        return -1, -1
+    detectors = ["Q1", "Q2", "U1", "U2"]
+    if not detector.upper() in detectors:
+        raise ValueError(f"ERROR: Detector %s does not exist")
+
+    detector = detector.upper()
     
     # Data type exists?
-    data_types = ['PWR', 'DEM', 'pwr', 'dem']
-    if (data_type in data_types) == False:
-        msg = 'ERROR: Data type %s does not exist' % data_type
-        print(msg)
-        return -1, -1
-    data_type.upper()
-    
-    channel = 'POL_%s%s' % (module, polarimeter)       
+    data_types = ["PWR", "DEM"]
+    if not data_type.upper() in data_types:
+        raise ValueError(f"ERROR: Data type %s does not exist")
+
+    data_type = data_type.upper()
+
+    channel = "POL_%s" % (polarimeter)
```

Comment:
Better to be more generic:

```suggestion
    channel = f"POL_{polarimeter}"
```